node/src/accountant/dao_utils.rs

```diff
@@ -0,0 +1,479 @@
+                    limit_clause: "",
+                }),
+                Self::set_up_age_constrains(min_age, max_age, timestamp)
+                    .into_iter()
+                    .chain(Self::set_up_wei_constrains(vec![min_amount, max_amount]))
+                    .collect::<Vec<(&str, Box<dyn ToSql>)>>(),
+            ),
+        };
+        let accounts = Self::execute_query(conn, &finalized_stm, params, value_fetcher);
+        (!accounts.is_empty()).then_some(accounts)
+    }
+
+    fn execute_query<'a, R, F1>(
+        conn: &'a dyn ConnectionWrapper,
+        stm: &'a str,
+        params: RusqliteParamsWithOwnedToSql,
+        value_fetcher: F1,
+    ) -> Vec<R>
+    where
+        F1: Fn(&Row) -> rusqlite::Result<R>,
+    {
+        conn.prepare(stm)
+            .expect("select statement is wrong")
+            .query_map(
+                &*params
+                    .iter()
+                    .map(|(param_name, value)| (*param_name, value.as_ref()))
+                    .collect::<Vec<_>>(),
+                value_fetcher,
+            )
+            .unwrap_or_else(|e| panic!("database corrupt: {}", e))
+            .flatten()
+            .collect::<Vec<R>>()
+    }
+
+    fn set_up_age_constrains(
+        min_age: u64,
+        max_age: u64,
+        timestamp: SystemTime,
+    ) -> RusqliteParamsWithOwnedToSql {
+        let now = to_time_t(timestamp);
+        let to_time_t = |limit| now - checked_conversion::<u64, i64>(limit);
+        vec![
+            (":min_timestamp", Box::new(to_time_t(max_age))),
+            (":max_timestamp", Box::new(to_time_t(min_age))),
+        ]
+    }
+
+    fn set_up_wei_constrains(two_num_limits: Vec<N>) -> RusqliteParamsWithOwnedToSql
+    where
+        i128: From<N>,
+    {
+        [
+            (":min_balance_high_b", ":min_balance_low_b"),
+            (":max_balance_high_b", ":max_balance_low_b"),
+        ]
+        .into_iter()
+        .zip(two_num_limits.into_iter())
+        .flat_map(|(param_names, gwei_num)| {
+            let wei_num = i128::from(gwei_num) * WEIS_OF_GWEI;
+            let (high_bytes, low_bytes) = BigIntDivider::deconstruct(wei_num);
+            vec![
+                (param_names.0, Box::new(high_bytes) as Box<dyn ToSql>),
+                (param_names.1, Box::new(low_bytes)),
+            ]
+        })
+        .collect()
+    }
+
+    fn ordering(
+        ordering: TopRecordsOrdering,
+        age_param: &'static str,
+    ) -> (&'static str, &'static str) {
+        match ordering {
+            TopRecordsOrdering::Age => (age_param, "balance_high_b desc, balance_low_b desc"),
+            TopRecordsOrdering::Balance => ("balance_high_b desc, balance_low_b desc", age_param),
+        }
+    }
+}
+
+impl<T: Copy> From<&RangeQuery<T>> for CustomQuery<T> {
+    fn from(user_input: &RangeQuery<T>) -> Self {
+        Self::RangeQuery {
+            min_age_s: user_input.min_age_s,
```

Comment:
Think about whether it would make the code in general more or less complex to have CustomQuery::RangeQuery have two fields, one of type RangeQuery and one of type SystemTime.  I don't know the answer; I just pose the question.